modules/scheduler/batch-job-template/main.tf

```diff
@@ -94,3 +94,17 @@ resource "local_file" "job_template" {
   content  = local.job_template_contents
   filename = local.job_template_output_path
 }
+
+resource "null_resource" "submit_job" {
+  depends_on = [ local_file.job_template ]
+
+  # A new deployment should always submit a new job. Old finished jobs aren't persistent parts of
+  # Cloud infrastructure.
+  triggers = {
+    always_run = timestamp()
+  }
+
+  provisioner "local-exec" {
+    command = var.submit ? "gcloud batch jobs submit --project=${var.project_id} --location=${var.region} --config=${local_file.job_template.filename}" : ""
```

Comment:
The error would appear on the console and could be a full USAGE message from gcloud, if the arguments were wrong, or an "Unable to parse config" message if there was something wrong with the job spec YAML file. Or it could be a message from the service itself, like if the Batch API isn't enabled on the target project. There are lots of possibilities, I guess.

Here are some examples where I deliberately caused an error:

```
ERROR: (gcloud.batch.jobs.submit) argument --config: Unable to read file [baz]: [Errno 2] No such file or directory: 'baz'
Usage: gcloud batch jobs submit [[JOB] --location=LOCATION] (--config=CONFIG --container-commands-file=CONTAINER_COMMANDS_FILE --container-entrypoint=CONTAINER_ENTRYPOINT --container-image-uri=CONTAINER_IMAGE_URI | --script-file-path=SCRIPT_FILE_PATH | --script-text=SCRIPT_TEXT) [optional flags]
  optional flags may be  --config | --container-commands-file |
                         --container-entrypoint | --container-image-uri |
                         --no-external-ip-address | --help | --job-prefix |
                         --location | --machine-type | --network | --priority |
                         --provisioning-model | --script-file-path |
                         --script-text | --subnetwork

For detailed information on this command and its flags, run:
  gcloud batch jobs submit --help
```

```
ERROR: (gcloud.batch.jobs.submit) Unable to parse config file: Expecting value: line 1 column 1 (char 0)
```

```
ERROR: (gcloud.batch.jobs.submit) CANCELLED: Invalid project state: Project disabled, satisfaction error code: INVALID_ARGUMENT
```